ecl/hql/hqltrans.cpp

```diff
@@ -4564,28 +4564,50 @@ void ScopedDependentTransformer::pushChildContext(IHqlExpression * expr, IHqlExp
     //NB: For this transformer it is the tables that are in scope that matter
     IHqlExpression * scope = expr->queryNormalizedSelector(false);
     //NB: Do no call createComma because that calls createDataset which unwinds a comma list!
+    bool identical = false;
     if (childScope)
-        childScope.setown(createValue(no_comma,LINK(scope), childScope.getClear()));
-    else
-        childScope.set(scope);
+    {
+        IHqlExpression * cur = childScope;
+        while (cur->getOperator() == no_comma)
+        {
+            IHqlExpression * prev = cur->queryChild(0);
+            if (prev == scope)
+                identical = true;
```

Comment:
@ghalliday Not sure why the loop is continuing once identical is set to 'true'..could it end the loop once identical==true?